chain/chain/src/chain.rs

```diff
@@ -1643,7 +1643,12 @@ impl Chain {
         let header = self.get_block_header(&sync_hash)?;
         let hash = *header.prev_hash();
         let prev_block = self.get_block(&hash)?;
-        let new_tail = prev_block.header().height();
+        let mut new_tail = prev_block.header().height() - 2;
+        if let Some(min_height_included) =
+            prev_block.chunks().iter().map(|chunk| chunk.height_included()).min()
```

Comment:
We probably can just unwrap here, because block must have all chunks headers, possibly copied.